### ISSUE TITLE
Runtime error with latest commit 0a5a7ee

### ISSUE THREAD

**Issue by jharwinbarrozo** [daniestevez/gr-satellites#176]
$ gr_satellites FUNcube-1 --wavfile ./satellite-recordings/ao73.wav --samp_rate 48e3 
Could not find port: out in:
system

Traceback (most recent call last):
  File "/usr/local/bin/gr_satellites", line 214, in <module>
    main()
  File "/usr/local/bin/gr_satellites", line 200, in main
    tb = gr_satellites_top_block(parser)
  File "/usr/local/bin/gr_satellites", line 107, in __init__
    self.flowgraph = satellites.core.gr_satellites_flowgraph(samp_rate = options.samp_rate, iq = self.options.iq,
  File "/usr/local/lib/python3/dist-packages/satellites/core/gr_satellites_flowgraph.py", line 164, in __init__
    deframer = self.get_deframer(transmitter['framing'])(options = options, **deframer_additional_options)
  File "/usr/local/lib/python3/dist-packages/satellites/components/deframers/ao40_fec_deframer.py", line 53, in __init__
    self.msg_connect((self.deframer, 'out'), (self.deinterleaver, 'in'))
  File "/usr/lib/python3/dist-packages/gnuradio/gr/hier_block2.py", line 60, in wrapped
    func(self, src.to_basic_block(), srcport, dst.to_basic_block(), dstport)
  File "/usr/lib/python3/dist-packages/gnuradio/gr/hier_block2.py", line 132, in msg_connect
    self.primitive_msg_connect(*args)
  File "/usr/lib/python3/dist-packages/gnuradio/gr/runtime_swig.py", line 3006, in primitive_msg_connect
    return _runtime_swig.hier_block2_sptr_primitive_msg_connect(self, *args)
RuntimeError: invalid msg port in connect() or disconnect()

Git checking out to 3.5.0 release https://github.com/daniestevez/gr-satellites/commit/b8842b3ce77a93363180336ed93f7151a373de9c and basically re-compiling solves the issue.

**Comment by 5ch4um1** [comment]
Which version of gnuradio are you using, and how did you install it?
I just followed the install instructions and could not reproduce your error:

$ gr_satellites FUNcube-1 --wavfile ./satellite-recordings/ao73.wav --samp_rate 48e3
Frame type WO10
----------------------------------------
Realtime telemetry:
----------------------------------------
Container: 
    eps = Container: 
        photovoltage = ListContainer: 
            0
            0
            0
        photocurrent = 0
        batteryvoltage = 8140
        systemcurrent = 206
        rebootcount = 721
        softwareerrors = 0
        boostconvertertemp = ListContainer: 

and so on...

$ gr_satellites --version
gr_satellites v3.6.0-git

**Comment by daniestevez** [comment]
I can't reproduce the error either, and I can't think of any reason for the changes between https://github.com/daniestevez/gr-satellites/commit/b8842b3ce77a93363180336ed93f7151a373de9c and https://github.com/daniestevez/gr-satellites/commit/0a5a7ee33082ceb471c44a4343c8c5e40a2be161 to cause this.

**Comment by jharwinbarrozo** [comment]
I installed my gnuradio via ubuntu ppa installation, i believe it is version 3.8.2
`sudo add-apt-repository ppa:gnuradio/gnuradio-releases`
`sudo apt-get update`
`sudo apt install gnuradio`

**Comment by daniestevez** [comment]
What does `ctest -V` inside the `build` directory report with the commit that doesn't work for you?

**Comment by jharwinbarrozo** [comment]
> What does `ctest -V` inside the `build` directory report with the commit that doesn't work for you?

Hi Daniel,

Here's the full log from cmake --> make ---> make test > ctest -V

jharwin@jharwin-ubuntu://home/jharwin/test/gr-satellites/$ git log -1
commit 0a5a7ee33082ceb471c44a4343c8c5e40a2be161 (HEAD -> master, origin/master, origin/HEAD)
Author: Daniel Estévez <daniel@destevez.net>
Date:   Thu Oct 8 19:54:07 2020 +0200

    Update NORAD ID for DEKART
    
    See #169
jharwin@jharwin-ubuntu:/home/jharwin/test/gr-satellites$ mkdir build
jharwin@jharwin-ubuntu:/home/jharwin/test/gr-satellites$ cd build
jharwin@jharwin-ubuntu/home/jharwin/test/gr-satellites/build$ cmake ..
-- The CXX compiler identification is GNU 9.3.0
-- The C compiler identification is GNU 9.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Build type not specified: defaulting to release.
-- Found LOG4CPP: /usr/lib/x86_64-linux-gnu/liblog4cpp.so
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'gmp'
--   Found gmp, version 6.2.0
-- Found GMP: /usr/lib/x86_64-linux-gnu/libgmpxx.so  
-- Checking for module 'mpir >= 3.0'
--   No package 'mpir' found
-- Could NOT find MPIR (missing: MPIRXX_LIBRARY MPIR_LIBRARY MPIR_INCLUDE_DIR) 
-- Found MPLIB: /usr/lib/x86_64-linux-gnu/libgmpxx.so  
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found suitable version "1.71.0", minimum required is "1.71.0") found components: date_time program_options filesystem system regex thread unit_test_framework 
-- Found VOLK: Volk::volk  
-- User set python executable /usr/bin/python3
-- Found PythonInterp: /usr/bin/python3 (found version "3.8.5") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so (found suitable exact version "3.8.5") 
-- Found Git: /usr/bin/git  
-- Extracting version information from git describe...
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found components: doxygen missing components: dot
-- Using install prefix: /usr/local
-- Building for version: v3-alpha4-294-g0a5a7ee3 / 3.6.0git
-- No C++ unit tests... skipping
-- Checking for module SWIG
-- Found SWIG version 4.0.1.
-- Found SWIG: /usr/bin/swig4.0  
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so (found version "3.8.5") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jharwin/test/gr-satellites/build
jharwin@jharwin-ubuntu:/home/jharwin/test/gr-satellites/build$ make
Scanning dependencies of target gnuradio-satellites
[  1%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/decode_rs_impl.cc.o
[  2%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/encode_rs_impl.cc.o
[  4%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/ao40_rs_decoder_impl.cc.o
[  5%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/ax100_decode_impl.cc.o
[  6%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/u482c_decode_impl.cc.o
[  8%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/randomizer.c.o
[  9%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/golay24.c.o
[ 10%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/lilacsat1_demux_impl.cc.o
[ 12%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/varlen_packet_tagger_impl.cc.o
[ 13%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/varlen_packet_framer_impl.cc.o
[ 14%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/viterbi.c.o
[ 16%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/nusat_decoder_impl.cc.o
[ 17%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/descrambler308_impl.cc.o
[ 18%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/decode_rs_general_impl.cc.o
[ 20%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/decode_rs_interleaved_impl.cc.o
[ 21%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/distributed_syncframe_soft_impl.cc.o
[ 22%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/matrix_deinterleaver_soft_impl.cc.o
[ 24%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/radecoder/ra_config.c.o
[ 25%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/radecoder/ra_decoder_gen.c.o
[ 26%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/radecoder/ra_lfsr.c.o
[ 28%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/radecoder/ra_encoder.c.o
[ 29%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/decode_ra_code_impl.cc.o
[ 30%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/nrzi_encode_impl.cc.o
[ 32%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/nrzi_decode_impl.cc.o
[ 33%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/decode_rs_8.c.o
[ 34%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/decode_rs_ccsds.c.o
[ 36%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/decode_rs_char.c.o
[ 37%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/encode_rs_8.c.o
[ 38%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/encode_rs_ccsds.c.o
[ 40%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/encode_rs_char.c.o
[ 41%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/taltab.c.o
[ 42%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/ccsds.c.o
[ 44%] Building C object lib/CMakeFiles/gnuradio-satellites.dir/libfec/init_rs_char.c.o
[ 45%] Linking CXX shared library libgnuradio-satellites.so
[ 45%] Built target gnuradio-satellites
Scanning dependencies of target pygen_apps_8d80d
[ 46%] Shebangin gr_satellites
[ 48%] Shebangin jy1sat_ssdv
[ 49%] Shebangin smog_p_spectrum
[ 49%] Built target pygen_apps_8d80d
Scanning dependencies of target generate_manpage_smog_p_spectrum.1
[ 49%] Built target generate_manpage_smog_p_spectrum.1
Scanning dependencies of target generate_manpage_jy1sat_ssdv.1
[ 49%] Built target generate_manpage_jy1sat_ssdv.1
Scanning dependencies of target generate_manpage_gr_satellites.1
[ 49%] Built target generate_manpage_gr_satellites.1
Scanning dependencies of target doxygen_target
[ 50%] Generating documentation with doxygen
warning: Tag 'PERL_PATH' at line 1686 of file '/home/jharwin/test/gr-satellites/build/docs/doxygen/Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'MSCGEN_PATH' at line 1707 of file '/home/jharwin/test/gr-satellites/build/docs/doxygen/Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
[ 50%] Built target doxygen_target
Scanning dependencies of target _satellites_swig_doc_tag
[ 52%] Building CXX object swig/CMakeFiles/_satellites_swig_doc_tag.dir/_satellites_swig_doc_tag.cpp.o
[ 53%] Linking CXX executable _satellites_swig_doc_tag
[ 53%] Built target _satellites_swig_doc_tag
Scanning dependencies of target satellites_swig_swig_doc
[ 54%] Generating doxygen xml for satellites_swig_doc docs
warning: Tag 'PERL_PATH' at line 1654 of file '/home/jharwin/test/gr-satellites/build/swig/satellites_swig_doc_swig_docs/Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'MSCGEN_PATH' at line 1675 of file '/home/jharwin/test/gr-satellites/build/swig/satellites_swig_doc_swig_docs/Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
[ 56%] Generating python docstrings for satellites_swig_doc
[ 56%] Built target satellites_swig_swig_doc
Scanning dependencies of target satellites_swig_swig_compilation
[ 57%] Swig source satellites_swig.i
[ 57%] Built target satellites_swig_swig_compilation
Scanning dependencies of target satellites_swig
[ 58%] Building CXX object swig/CMakeFiles/satellites_swig.dir/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx.o
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function ‘PyTypeObject* SwigPyObject_TypeOnce()’:
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1770:5: warning: missing initializer for member ‘_typeobject::tp_vectorcall’ [-Wmissing-field-initializers]
 1770 |     };
      |     ^
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1770:5: warning: missing initializer for member ‘_typeobject::tp_print’ [-Wmissing-field-initializers]
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function ‘PyTypeObject* SwigPyPacked_TypeOnce()’:
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1931:5: warning: missing initializer for member ‘_typeobject::tp_vectorcall’ [-Wmissing-field-initializers]
 1931 |     };
      |     ^
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1931:5: warning: missing initializer for member ‘_typeobject::tp_print’ [-Wmissing-field-initializers]
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function ‘PyTypeObject* swig_varlink_type()’:
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:65013:7: warning: missing initializer for member ‘_typeobject::tp_vectorcall’ [-Wmissing-field-initializers]
65013 |       };
      |       ^
/home/jharwin/test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:65013:7: warning: missing initializer for member ‘_typeobject::tp_print’ [-Wmissing-field-initializers]
[ 60%] Linking CXX shared module _satellites_swig.so
[ 60%] Built target satellites_swig
Scanning dependencies of target pygen_swig_f34ea
[ 61%] Generating satellites_swig.pyc
[ 62%] Generating satellites_swig.pyo
[ 62%] Built target pygen_swig_f34ea
Scanning dependencies of target pygen_python_2aad9
[ 64%] Generating __init__.pyc, kiss.pyc, hdlc.pyc, kiss_to_pdu.pyc, pdu_to_kiss.pyc, hdlc_framer.pyc, hdlc_deframer.pyc, check_address.pyc, fixedlen_tagger.pyc, csp_header.pyc, print_header.pyc, crc32c.pyc, check_crc.pyc, swap_crc.pyc, swap_header.pyc, submit.pyc, print_timestamp.pyc, adsb_kml.pyc, ks1q_header_remover.pyc, ao40_uncoded_crc.pyc, check_ao40_uncoded_crc.pyc, lilacsat1_gps_kml.pyc, check_tt64_crc.pyc, append_crc32c.pyc, strip_ax25_header.pyc, snet_deframer.pyc, bch15.pyc, beesat_classifier.pyc, adapters.pyc, snet_classifier.pyc, k2sat_deframer.pyc, feh.pyc, cc11xx_packet_crop.pyc, check_cc11xx_crc.pyc, cc11xx_remove_length.pyc, pwsat2_telemetry_parser.pyc, pwsat2_submitter.pyc, eseo_packet_crop.pyc, eseo_line_decoder.pyc, check_eseo_crc.pyc, funcube_submit.pyc, reflect_bytes.pyc, check_astrocast_crc.pyc, ecss_pus.pyc, swiatowid_packet_crop.pyc, check_swiatowid_crc.pyc, swiatowid_packet_split.pyc, manchester_sync.pyc, header_remover.pyc, aausat4_remove_fsm.pyc, ngham_packet_crop.pyc, ngham_remove_padding.pyc, ngham_check_crc.pyc, bme_submitter.pyc
[ 65%] Generating __init__.pyo, kiss.pyo, hdlc.pyo, kiss_to_pdu.pyo, pdu_to_kiss.pyo, hdlc_framer.pyo, hdlc_deframer.pyo, check_address.pyo, fixedlen_tagger.pyo, csp_header.pyo, print_header.pyo, crc32c.pyo, check_crc.pyo, swap_crc.pyo, swap_header.pyo, submit.pyo, print_timestamp.pyo, adsb_kml.pyo, ks1q_header_remover.pyo, ao40_uncoded_crc.pyo, check_ao40_uncoded_crc.pyo, lilacsat1_gps_kml.pyo, check_tt64_crc.pyo, append_crc32c.pyo, strip_ax25_header.pyo, snet_deframer.pyo, bch15.pyo, beesat_classifier.pyo, adapters.pyo, snet_classifier.pyo, k2sat_deframer.pyo, feh.pyo, cc11xx_packet_crop.pyo, check_cc11xx_crc.pyo, cc11xx_remove_length.pyo, pwsat2_telemetry_parser.pyo, pwsat2_submitter.pyo, eseo_packet_crop.pyo, eseo_line_decoder.pyo, check_eseo_crc.pyo, funcube_submit.pyo, reflect_bytes.pyo, check_astrocast_crc.pyo, ecss_pus.pyo, swiatowid_packet_crop.pyo, check_swiatowid_crc.pyo, swiatowid_packet_split.pyo, manchester_sync.pyo, header_remover.pyo, aausat4_remove_fsm.pyo, ngham_packet_crop.pyo, ngham_remove_padding.pyo, ngham_check_crc.pyo, bme_submitter.pyo
[ 65%] Built target pygen_python_2aad9
Scanning dependencies of target pygen_python_ccsds_eef07
[ 66%] Generating __init__.pyc, pathID_demultiplexer.pyc, space_packet.pyc, space_packet_parser.pyc, space_packet_primaryheader_adder.pyc, space_packet_time_stamp_adder.pyc, telecommand.pyc, telecommand_parser.pyc, telecommand_primaryheader_adder.pyc, telemetry.pyc, telemetry_ocf_adder.pyc, telemetry_packet_reconstruction.pyc, telemetry_parser.pyc, telemetry_primaryheader_adder.pyc, virtual_channel_demultiplexer.pyc
[ 68%] Generating __init__.pyo, pathID_demultiplexer.pyo, space_packet.pyo, space_packet_parser.pyo, space_packet_primaryheader_adder.pyo, space_packet_time_stamp_adder.pyo, telecommand.pyo, telecommand_parser.pyo, telecommand_primaryheader_adder.pyo, telemetry.pyo, telemetry_ocf_adder.pyo, telemetry_packet_reconstruction.pyo, telemetry_parser.pyo, telemetry_primaryheader_adder.pyo, virtual_channel_demultiplexer.pyo
[ 68%] Built target pygen_python_ccsds_eef07
Scanning dependencies of target pygen_python_components_8eada
[ 69%] Generating __init__.pyc
[ 70%] Generating __init__.pyo
[ 70%] Built target pygen_python_components_8eada
Scanning dependencies of target pygen_python_components_deframers_dabc8
[ 72%] Generating __init__.pyc, aausat4_deframer.pyc, ao40_fec_deframer.pyc, ao40_uncoded_deframer.pyc, astrocast_9k6_deframer.pyc, astrocast_fx25_deframer.pyc, ax100_deframer.pyc, ax25_deframer.pyc, ccsds_concatenated_deframer.pyc, ccsds_rs_deframer.pyc, eseo_deframer.pyc, k2sat_deframer.pyc, lilacsat_1_deframer.pyc, lucky7_deframer.pyc, mobitex_deframer.pyc, ngham_deframer.pyc, nusat_deframer.pyc, ops_sat_deframer.pyc, reaktor_hello_world_deframer.pyc, sat_3cat_1_deframer.pyc, smogp_ra_deframer.pyc, smogp_signalling_deframer.pyc, snet_deframer.pyc, swiatowid_deframer.pyc, tt64_deframer.pyc, u482c_deframer.pyc, ua01_deframer.pyc
[ 73%] Generating __init__.pyo, aausat4_deframer.pyo, ao40_fec_deframer.pyo, ao40_uncoded_deframer.pyo, astrocast_9k6_deframer.pyo, astrocast_fx25_deframer.pyo, ax100_deframer.pyo, ax25_deframer.pyo, ccsds_concatenated_deframer.pyo, ccsds_rs_deframer.pyo, eseo_deframer.pyo, k2sat_deframer.pyo, lilacsat_1_deframer.pyo, lucky7_deframer.pyo, mobitex_deframer.pyo, ngham_deframer.pyo, nusat_deframer.pyo, ops_sat_deframer.pyo, reaktor_hello_world_deframer.pyo, sat_3cat_1_deframer.pyo, smogp_ra_deframer.pyo, smogp_signalling_deframer.pyo, snet_deframer.pyo, swiatowid_deframer.pyo, tt64_deframer.pyo, u482c_deframer.pyo, ua01_deframer.pyo
[ 73%] Built target pygen_python_components_deframers_dabc8
Scanning dependencies of target pygen_python_components_demodulators_3096b
[ 74%] Generating __init__.pyc, afsk_demodulator.pyc, bpsk_demodulator.pyc, fsk_demodulator.pyc
[ 76%] Generating __init__.pyo, afsk_demodulator.pyo, bpsk_demodulator.pyo, fsk_demodulator.pyo
[ 76%] Built target pygen_python_components_demodulators_3096b
Scanning dependencies of target pygen_python_components_datasinks_2dadd
[ 77%] Generating __init__.pyc, codec2_udp_sink.pyc, file_receiver.pyc, hexdump_sink.pyc, kiss_file_sink.pyc, kiss_server_sink.pyc, telemetry_parser.pyc, telemetry_submit.pyc
[ 78%] Generating __init__.pyo, codec2_udp_sink.pyo, file_receiver.pyo, hexdump_sink.pyo, kiss_file_sink.pyo, kiss_server_sink.pyo, telemetry_parser.pyo, telemetry_submit.pyo
[ 78%] Built target pygen_python_components_datasinks_2dadd
Scanning dependencies of target pygen_python_components_datasources_8dac9
[ 80%] Generating __init__.pyc, kiss_file_source.pyc
[ 81%] Generating __init__.pyo, kiss_file_source.pyo
[ 81%] Built target pygen_python_components_datasources_8dac9
Scanning dependencies of target pygen_python_components_transports_85acb
[ 82%] Generating __init__.pyc, kiss_transport.pyc
[ 84%] Generating __init__.pyo, kiss_transport.pyo
[ 84%] Built target pygen_python_components_transports_85acb
Scanning dependencies of target pygen_python_core_6dee2
[ 85%] Generating __init__.pyc, gr_satellites_flowgraph.pyc
[ 86%] Generating __init__.pyo, gr_satellites_flowgraph.pyo
[ 86%] Built target pygen_python_core_6dee2
Scanning dependencies of target pygen_python_hier_363e4
[ 88%] Generating __init__.pyc, ccsds_descrambler.pyc, ccsds_viterbi.pyc, si4463_scrambler.pyc, sync_to_pdu.pyc, sync_to_pdu_packed.pyc, sync_to_pdu_soft.pyc, rms_agc.pyc, rms_agc_f.pyc, pn9_scrambler.pyc
[ 89%] Generating __init__.pyo, ccsds_descrambler.pyo, ccsds_viterbi.pyo, si4463_scrambler.pyo, sync_to_pdu.pyo, sync_to_pdu_packed.pyo, sync_to_pdu_soft.pyo, rms_agc.pyo, rms_agc_f.pyo, pn9_scrambler.pyo
[ 89%] Built target pygen_python_hier_363e4
Scanning dependencies of target pygen_python_filereceiver_36a34
[ 90%] Generating __init__.pyc, by70_1.pyc, dsat.pyc, k2sat.pyc, filereceiver.pyc, imagereceiver.pyc, sat_1kuns_pf.pyc, smogp.pyc, swiatowid.pyc
[ 92%] Generating __init__.pyo, by70_1.pyo, dsat.pyo, k2sat.pyo, filereceiver.pyo, imagereceiver.pyo, sat_1kuns_pf.pyo, smogp.pyo, swiatowid.pyo
[ 92%] Built target pygen_python_filereceiver_36a34
Scanning dependencies of target pygen_python_satyaml_36973
[ 93%] Generating __init__.pyc, satyaml.pyc
[ 94%] Generating __init__.pyo, satyaml.pyo
[ 94%] Built target pygen_python_satyaml_36973
Scanning dependencies of target pygen_python_telemetry_830da
[ 96%] Generating __init__.pyc, aausat4.pyc, amicalsat.pyc, au03.pyc, ax25.pyc, by02.pyc, by70_1.pyc, csp.pyc, dstar_one.pyc, eseo.pyc, floripasat.pyc, funcube.pyc, gomx_1.pyc, gomx_3.pyc, kr01.pyc, lume.pyc, mysat1.pyc, picsat.pyc, qo100.pyc, quetzal1.pyc, sat_1kuns_pf.pyc, sat_3cat_1.pyc, sat_3cat_2.pyc, smogp.pyc, snet.pyc, suomi100.pyc, trisat.pyc, upmsat_2.pyc
[ 97%] Generating __init__.pyo, aausat4.pyo, amicalsat.pyo, au03.pyo, ax25.pyo, by02.pyo, by70_1.pyo, csp.pyo, dstar_one.pyo, eseo.pyo, floripasat.pyo, funcube.pyo, gomx_1.pyo, gomx_3.pyo, kr01.pyo, lume.pyo, mysat1.pyo, picsat.pyo, qo100.pyo, quetzal1.pyo, sat_1kuns_pf.pyo, sat_3cat_1.pyo, sat_3cat_2.pyo, smogp.pyo, snet.pyo, suomi100.pyo, trisat.pyo, upmsat_2.pyo
[ 97%] Built target pygen_python_telemetry_830da
Scanning dependencies of target pygen_python_utils_5ddb6
[ 98%] Generating __init__.pyc, config.pyc, options_block.pyc
[100%] Generating __init__.pyo, config.pyo, options_block.pyo
[100%] Built target pygen_python_utils_5ddb6
jharwin@jharwin-ubuntu:~/test/gr-satellites/build$ make test
Running tests...
Test project /home/jharwin/test/gr-satellites/build
    Start 1: qa_hdlc
1/7 Test #1: qa_hdlc ..........................   Passed    0.30 sec
    Start 2: qa_kiss
2/7 Test #2: qa_kiss ..........................   Passed    0.27 sec
    Start 3: qa_nrzi
3/7 Test #3: qa_nrzi ..........................   Passed    0.27 sec
    Start 4: qa_ao40_fec_deframer
4/7 Test #4: qa_ao40_fec_deframer .............   Passed    0.37 sec
    Start 5: qa_telemetry_parser
5/7 Test #5: qa_telemetry_parser ..............   Passed    0.27 sec
    Start 6: qa_satyaml
6/7 Test #6: qa_satyaml .......................   Passed    0.26 sec
    Start 7: qa_by02
7/7 Test #7: qa_by02 ..........................   Passed    0.25 sec

100% tests passed, 0 tests failed out of 7

Total Test time (real) =   1.97 sec
jharwin@jharwin-ubuntu:/home/jharwin/test/gr-satellites/build$ ctest -V
UpdateCTestConfiguration  from :/home/jharwin/test/gr-satellites/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/jharwin/test/gr-satellites/build/DartConfiguration.tcl
Test project /home/jharwin/test/gr-satellites/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: qa_hdlc

1: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/qa_hdlc_test.sh"
1: Test timeout computed to be: 10000000
1: .
1: ----------------------------------------------------------------------
1: Ran 1 test in 0.013s
1: 
1: OK
1/7 Test #1: qa_hdlc ..........................   Passed    0.31 sec
test 2
    Start 2: qa_kiss

2: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/qa_kiss_test.sh"
2: Test timeout computed to be: 10000000
2: .
2: ----------------------------------------------------------------------
2: Ran 1 test in 0.005s
2: 
2: OK
2/7 Test #2: qa_kiss ..........................   Passed    0.27 sec
test 3
    Start 3: qa_nrzi

3: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/qa_nrzi_test.sh"
3: Test timeout computed to be: 10000000
3: ...
3: ----------------------------------------------------------------------
3: Ran 3 tests in 0.003s
3: 
3: OK
3/7 Test #3: qa_nrzi ..........................   Passed    0.27 sec
test 4
    Start 4: qa_ao40_fec_deframer

4: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/components/deframers/qa_ao40_fec_deframer_test.sh"
4: Test timeout computed to be: 10000000
4: .
4: ----------------------------------------------------------------------
4: Ran 1 test in 0.019s
4: 
4: OK
4/7 Test #4: qa_ao40_fec_deframer .............   Passed    0.37 sec
test 5
    Start 5: qa_telemetry_parser

5: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/components/datasinks/qa_telemetry_parser_test.sh"
5: Test timeout computed to be: 10000000
5: .
5: ----------------------------------------------------------------------
5: Ran 1 test in 0.001s
5: 
5: OK
5/7 Test #5: qa_telemetry_parser ..............   Passed    0.27 sec
test 6
    Start 6: qa_satyaml

6: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/satyaml/qa_satyaml_test.sh"
6: Test timeout computed to be: 10000000
6: .
6: ----------------------------------------------------------------------
6: Ran 1 test in 0.192s
6: 
6: OK
6/7 Test #6: qa_satyaml .......................   Passed    0.26 sec
test 7
    Start 7: qa_by02

7: Test command: /usr/bin/sh "/home/jharwin/test/gr-satellites/build/python/telemetry/qa_by02_test.sh"
7: Test timeout computed to be: 10000000
7: .
7: ----------------------------------------------------------------------
7: Ran 1 test in 0.000s
7: 
7: OK
7/7 Test #7: qa_by02 ..........................   Passed    0.23 sec

100% tests passed, 0 tests failed out of 7

Total Test time (real) =   1.98 sec

**Comment by daniestevez** [comment]
So all your tests seem to pass, including `qa_ao40_fec_deframer_test`, which is a test of the AO-40 FEC deframer block. Interestingly, the failure you got when running `gr_satellites FUNcube-1 --wavfile ./satellite-recordings/ao73.wav --samp_rate 48e3` was in the constructor of `ao40_fec_deframer`. But the test is running properly. I wonder what is happening.

Maybe you have a mixture of installations on your system, if you've been installing different versions of gr-satellites in your system. Perhaps it would be good to remove all trace of gr-satellites on your system and install again. The test runs directly from the `build` directory, so it wouldn't be affected by this.

**Comment by jharwinbarrozo** [comment]
When I installed the https://github.com/daniestevez/gr-satellites/commit/0a5a7ee33082ceb471c44a4343c8c5e40a2be161 I had it on a fresh ubuntu 20.04 clean slate, so I am pretty sure I don't have any gr-satellites installed anywhere. 

The directory where I ran the ctest -v was just a new git clone from the latest commit, but I didn't install it(sudo make install).

I am going to close this as I don't have any idea too what causes it, as the test were all okay.
